### DISCUSSION TITLE
Dark mode can't handle scanned pdf files

### DISCUSSION THREAD

**Discussion by anf4108** [James-Yu/LaTeX-Workshop#3885] [Q&A]
PDF viewers can open most of my files in dark mode and invert colors effectively, but it seems that they are unable to process scanned PDF files. For regular files, they are displayed like this:
<img width="506" alt="截屏2023-05-23 18 55 38" src="https://github.com/James-Yu/LaTeX-Workshop/assets/110650085/b4c72ba6-8f40-43d3-b048-6e08951ffb45">
For some scanned files, they are displayed like this:
<img width="416" alt="截屏2023-05-23 18 56 00" src="https://github.com/James-Yu/LaTeX-Workshop/assets/110650085/5142d8f6-2766-422e-8890-08596a675357">
And for others, they cannot be inverted at all:
<img width="532" alt="截屏2023-05-23 18 57 12" src="https://github.com/James-Yu/LaTeX-Workshop/assets/110650085/da258439-8d6a-4ca4-83ac-a7a6b1ced85b">
When I use other viewers, like Zathura with recolor on , it handles all of these cases effectively:
<img width="1126" alt="截屏2023-05-23 18 58 09" src="https://github.com/James-Yu/LaTeX-Workshop/assets/110650085/3c06bdc5-7e9d-476b-b204-c35f6bc79f34">
<img width="1069" alt="截屏2023-05-23 19 02 10" src="https://github.com/James-Yu/LaTeX-Workshop/assets/110650085/76aeec58-0303-47ad-a976-6005161a04bb">

I suspect that the reason could be that PDF viewers process images separately for colored PDFs, which leads to some PDFs being unable to be viewed in dark mode or with poor results. Differently, viewers like Zathura processes PDFs into grayscale before inverting colors.
However, the purpose of dark mode is to present files in a dark color scheme, so I hope there can be at least one alternative option that allows users to use dark mode in more situations, even if it's not fully compatible with color files.

**Comment by James-Yu** [comment]
Your request is related to pdf.js . I recommend you to file an issue at their repo.
